### PR TITLE
FLUT-7436 - [Others] UG documentation for DataGrid column drag and drop.

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
## Description ##

* The UG documentation for the DataGrid column drag and drop has been added.

### CODE REVIEW

File: Flutter/datagrid/column-drag-and-drop.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+        source: employeeDataSource,
+        allowColumnsDragging: true,
+        columns: columns,
+        onColumnDragging: (DataGridColumnDragDetails details) {
+          if (details.action == DataGridColumnDragAction.dropped &&
+              details.to != null) {
+            final GridColumn rearrangeColumn = columns[details.from];
+            columns.removeAt(details.from);
+            columns.insert(details.to!, rearrangeColumn);
+            employeeDataSource.buildDataGridRow();
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:81]: Rename the method **buildDataGridRows()**

File: Flutter/datagrid/column-drag-and-drop.md [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+            employeeDataSource.updateDtaGrid();
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:82]: Rename the method **refreshDataGrid()**

File: Flutter/datagrid/column-drag-and-drop.md [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+          columnName: 'salary',
+          label: Container(
+              padding: const EdgeInsets.all(8.0),
+              alignment: Alignment.center,
+              child: const Text('Salary'))),
+    ];
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:68]: build method and DataGridSource changes will be enough. Link the GitHub sample for the column drag-and-drop demo.

File: Flutter/datagrid/column-drag-and-drop.md [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+{% endtabs %}
+
+
+<img alt="Flutter datagrid shows a checkbox filter in web platform" src="images/column-drag-and-drop/column-drag-and-drop.gif" width="400"/>
+
+>**NOTE**:  
+To reorder the columns in the DataGrid, you should create an instance to hold the columns and then assign that instance to 
+the SfDataGrid.columns property instead of directly assigning the list of GridColumn. This allows you to reorder the collection within the callback, maintaining the desired column order.
+Additionally, it is important to build the rows based on the columns collection after reordering. This is necessary because
+the column index may change after the columns have been rearranged. By rebuilding the rows based on the updated columns collection, you ensure that the row data aligns correctly with the reordered columns.
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:153]: Provide bullet points.

File: Flutter/datagrid/column-drag-and-drop.md [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Flutter datagrid shows a checkbox filter in web platform" src="images/column-drag-and-drop/feedback-widget.gif" width="400"/>
+
+## Drag indicator customization
+
+The DataGrid allows you to customize the drag indicator by changing its color and thickness. The DataGrid should be wrapped inside the `SfDataGridTheme`.
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:254]: The color and thickness of the drag indicator can be customized by the SfDataGridThemeData.columnDragIndicatorColor and SfDataGridThemeData.columnDragIndicatorStrokeWidth properties.

File: Flutter/datagrid/column-drag-and-drop.md [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+
+{% tabs %}
+{% highlight Dart %}
+
+import 'package:syncfusion_flutter_core/theme.dart';
+
+{% endhighlight %}
+{% endtabs %}
+
+### Change the drag indicator color
```
Comment (ashok-kuvaraja) [inline Flutter/datagrid/column-drag-and-drop.md:266]: Don't need individual topics for color and thickness. Cover these topis inside the drag indicator customization itself and explain those in a single code snippets.

File: Flutter/datagrid/column-drag-and-drop.md (hunk 1 above)
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/column-drag-and-drop.md:81]: Renamed

File: Flutter/datagrid/column-drag-and-drop.md (hunk 2 above)
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/column-drag-and-drop.md:82]: Renamed.

File: Flutter/datagrid/column-drag-and-drop.md (hunk 6 above)
Comment (Tamilarasan-Paranthaman) [reply Flutter/datagrid/column-drag-and-drop.md:266]: Modified.

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Review by FarjanaParveen — APPROVED** [review]
This is fine

**Pull request merged**